api/mapping/forms.py

```diff
@@ -34,35 +35,65 @@ class ScanReportForm(forms.Form):
     )
     def clean_scan_report_file(self):
         if str(self.cleaned_data['scan_report_file']).endswith('.xlsx'):
-            xlsx = Xlsx2csv(self.cleaned_data['scan_report_file'], outputencoding="utf-8")
-
-            filepath = "/tmp/{}.csv".format(xlsx.workbook.sheets[0]["name"])
-            xlsx.convert(filepath)
-
-            with open(filepath, "rt") as f:
-                reader = csv.reader(f)
-                csv_header=next(reader)  # Get header row
-                set_header=['Table', 'Field', 'Description', 'Type', 'Max length', 'N rows', 'N rows checked', 'Fraction empty', 'N unique values', 'Fraction unique', 'Flag', 'Classification']
-                if set(set_header)==set(csv_header):
-                    for row in reader:
-                        flag_column=row[10]
-                        flag_column=flag_column.upper()
-                        classification_column=row[11]
-                        print(flag_column)
-                        if (flag_column in FLAG_CHOICES) or (flag_column==''):
-                            pass
-                        else:
-                            raise (forms.ValidationError( "Check Flag column values. Valid options are: {} or blank".format(list(FLAG_CHOICES.values()))))
-                        
-                        if (classification_column in VOCABULARY_CHOICES.values()) or (classification_column==''):
-                            pass
-                        else:
-                            raise (forms.ValidationError( "Check Classification column values. Valid options are:{} or blank".format(list(VOCABULARY_CHOICES.values()))))
-                    return self.cleaned_data['scan_report_file']
+
```

Comment:
swap this `if` for `if not` and then the `ValidationError` near the bottom can be dragged up here - reduces the indenting and is a little clearer in showing it will abort most of the function.

Reply:
This one still has the `else` at the bottom - change it to `if not` and then put the `ValidationError` in here.